tests/unit/api/multipartUpload.js

```diff
@@ -49,6 +55,26 @@ const initiateRequest = {
     headers: { host: `${bucketName}.s3.amazonaws.com` },
     url: `/${objectKey}?uploads`,
 };
+const retentionInitiateRequest = Object.assign(initiateRequest, {});
+retentionInitiateRequest.headers['x-amz-object-lock-mode'] = 'GOVERNANCE';
+retentionInitiateRequest.headers['x-amz-object-lock-retain-until-date'] =
+    futureDate;
+retentionInitiateRequest.bucketName = lockedBucket;
+const legalHoldInitiateRequest = Object.assign(initiateRequest, {});
+legalHoldInitiateRequest.headers['x-amz-object-lock-mode'] = 'GOVERNANCE';
+legalHoldInitiateRequest.headers['x-amz-object-lock-retain-until-date'] =
+    futureDate;
```

Comment:
( without using the spread operator this is what I came up with for this block... note that spread and object.assign does not merge recursively so you have to do two steps )
```suggestion
amzHeaders = {
  ['x-amz-object-lock-mode']: 'GOVERNANCE',
  ['x-amz-object-lock-retain-until-date']: futureData,
}
const legalHoldInitiateRequest = Object.assign(initiateRequest, {});
legalHoldInitiateRequest.headers = Object.assign(legalHoldInitiateRequest.headers, amzHeaders);
const retentionInitiateRequest = Object.assign(legalHoldInitiateRequest, {bucketName: lockedBucket});
```